### PR TITLE
implement Done command

### LINKED ISSUE #31: implement done command
- Duplicate and Modify UniqueTaskList to support Tasks that are done
- Rename it to be called UniqueDoneList

### PR DESCRIPTION
**Pull request by ZuoZhuolin**
This closes #31

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ This format can be used to add events held on a certain day.<br>
 This format can be used to add tasks with a stipulated deadline.<br>
 > Example: `add do lab homework on Friday 7pm` <br>
 -  `add TASK ITEM on [DATE]` <br>
-The use of word �on� gives more flexibility for you.<br>
+The use of word �on� gives more flexibility for you.<br>
```
Comment (marcusngwenjian) [inline docs/UserGuide.md]: Ensure that symbols do not turn into alien characters before creating a pull request.
image

File: src/main/java/seedu/taskell/model/task/TaskComplete.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        this.isCompleted = isCompleted;
+    }
+    
+    public String isCompleted() {
+        return isCompleted;
+    }
+    
+    public static boolean isValidTaskComplete(String TaskToValidate) {
+       return TaskToValidate == "finished"  || TaskToValidate == "not completed";           
+    }
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/model/task/TaskComplete.java:21]: Magic values. Please make "finished" and "not completed" as constants

File: src/test/java/seedu/taskell/testutil/TestTask.java [hunk 3]
```diff
@@ -44,7 +45,10 @@ public void setEndTime(TaskTime endTime) {
     public void setTaskDate(TaskDate taskDate) {
         this.taskDate = taskDate;
     }
-
+    
+    public void setTaskComplete(TaskComplete isCompleted){
+        this.isCompleted = isCompleted;
```
Comment (marcusngwenjian) [inline src/test/java/seedu/taskell/testutil/TestTask.java:50]: Add whitespace after bracket
image
image

File: src/test/java/seedu/taskell/testutil/TestUtil.java [hunk 4]
```diff
@@ -66,15 +66,15 @@ public static void assertThrows(Class<? extends Throwable> expected, Runnable ex
...
-                    new Task(new Description("George Best"), Task.EVENT_TASK, new TaskDate("1-1-2015"), new TaskTime("12:30AM"), new TaskTime("12:45AM"), new TaskPriority("0"), new UniqueTagList()),
-                    new Task(new Description("Hoon Meier"), Task.EVENT_TASK, new TaskDate("1-1-2015"), new TaskTime("12:30AM"), new TaskTime("12:45AM"), new TaskPriority("0"), new UniqueTagList()),
-                    new Task(new Description("Ida Mueller"), Task.EVENT_TASK, new TaskDate("1-1-2015"), new TaskTime("12:30AM"), new TaskTime("12:45AM"), new TaskPriority("0"), new UniqueTagList())
+                    new Task(new Description("Ali Muster"), Task.EVENT_TASK, new TaskDate("1-1-2015"), new TaskTime("12:30AM"), new TaskTime("12:45AM"), new TaskPriority("0"), new TaskComplete("not completed"), new UniqueTagList()),
+                    new Task(new Description("Boris Mueller"), Task.EVENT_TASK, new TaskDate("1-1-2015"), new TaskTime("12:30AM"), new TaskTime("12:45AM"), new TaskPriority("0"), new TaskComplete("not completed"), new UniqueTagList()),
```
Comment (marcusngwenjian) [inline src/test/java/seedu/taskell/testutil/TestUtil.java:70]: Magic value. Make "not completed" a constant in the TaskComplete class

File: src/main/java/seedu/taskell/model/task/TaskComplete.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package seedu.taskell.model.task;
+
+import java.util.Objects;
+
+public class TaskComplete {
+    
+    public static final String COMPLETED_INFO = "Completed";
+    public static final String NOT_COMPLETED_INFO = "Not Completed";
+    public final String isCompleted;
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/model/task/TaskComplete.java]: Refactor isComplete to "taskStatus". Variable starting with is/has/can/should are reserved for boolean variables.
image
image
image

**Comment by marcusngwenjian** [comment]
2 JUnit tests failed. Please revise.
image

**Comment by marcusngwenjian** [comment]
@ZuoZhuolin 
Todo: 
- Fix and add JUnit tests
- Resolve merge conflict

File: docs/UserGuide.md [hunk 6]
```diff
@@ -67,13 +67,15 @@ To view a list of all the tasks, <kbd>Enter</kbd> list.<br>
 <br>
 Formats: 
 - `list` <br>
-Prints a list of all the uncompleted tasks.<br>
+Prints a list of all tasks.<br>
```
Comment (Mamtha3005) [inline docs/UserGuide.md:70]: Change prints to displays

File: docs/UserGuide.md [hunk 7]
```diff
@@ -67,13 +67,15 @@ To view a list of all the tasks, <kbd>Enter</kbd> list.<br>
...
 - `list DATE` <br> 
 Prints a list of all the completed tasks.<br>
-- `list DONE` <br>
+- `listdone` <br>
```
Comment (Mamtha3005) [inline docs/UserGuide.md]: Better for the command o be list-done.
Similar to list-date, find-tag

File: docs/UserGuide.md [hunk 8]
```diff
@@ -67,13 +67,15 @@ To view a list of all the tasks, <kbd>Enter</kbd> list.<br>
...
 Prints a list of all the tasks due on the specific date.<br>
+- `listdo` <br>
```
Comment (Mamtha3005) [inline docs/UserGuide.md]: Isnt this alrd list command?

File: docs/UserGuide.md [hunk 9]
```diff
@@ -156,6 +162,7 @@ Find | `find KEYWORD [MORE_KEYWORDS]`
 Find Tag | `find-tag KEYWORD`
 List | `list`
 List Given Day | `list [DATE]`
-List Tasks Done | `list [DONE]`
+List Tasks Done | `listdone`
+List Tasks need to be done | `listdo`
```
Comment (Mamtha3005) [inline docs/UserGuide.md]: Why need to be done all lower caps??

File: src/main/java/seedu/taskell/logic/commands/DoneCommand.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import seedu.taskell.model.task.Task;
+import seedu.taskell.model.task.TaskComplete;
+import seedu.taskell.model.task.UniqueTaskList;
+import seedu.taskell.model.task.UniqueTaskList.TaskNotFoundException;
+
+public class DoneCommand extends Command {
+    public static final String COMMAND_WORD = "done";
+    
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Done the task identified by the index number used in the last task listing.\n"
```
Comment (Mamtha3005) [inline src/main/java/seedu/taskell/logic/commands/DoneCommand.java:15]: Marks the task as done vs Done

File: src/main/java/seedu/taskell/logic/commands/ListDoneCommand.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package seedu.taskell.logic.commands;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ListDoneCommand extends Command {
+    
+    public static final String COMMAND_WORD = "listdone";
```
Comment (Mamtha3005) [inline src/main/java/seedu/taskell/logic/commands/ListDoneCommand.java]: list-done

**Comment by Mamtha3005** [comment]
Documentation Reviewed. Please make the necessary changes.

**Pull request closed without merging**